### PR TITLE
Update to use requests library

### PR DESCRIPTION
**Pull request by gene1wood**
Switching to requests library
Adding test if response is JSON and return raw content if not

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -16,14 +15,14 @@ def read(fname):
...
     py_modules=['authzero'],
-    version="1.0.6",
-    author="Guillaume Destuynder",
-    author_email="gdestuynder@mozilla.com",
+    version="1.1.0",
+    author="Mozilla IAM",
+    author_email="gene@mozilla.com",
     description=("A super simple and basic client lib for Auth0"),
-    license="MPL",
+    license=" MPL-2.0",
```
Comment (floatingatoll) [inline setup.py:22]: There's extra whitespace here that may cause trouble for packaging sites?

**Review by floatingatoll — COMMENTED** [review]
@bkochendorfer Could you review this? It seems essentially fine to me.

File: setup.py (hunk 1 above)
Comment (gene1wood) [reply setup.py:22]: Whoops, good catch.

**Comment by gene1wood** [comment]
@bkochendorfer 
> I assume the error we saw previously was in trying to load a json blob in _handle_response which didn't exist.

Yes, that's right. The mitigation for the specific error we encountered is now to try to to parse JSON and then if it fails, just return the raw content.

File: authzero.py [hunk 2]
```diff
@@ -402,20 +385,22 @@ def get_access_token(self):
         self.access_token_scope = access_token.scope
         return access_token
 
-    def _request(self, rpath, rtype="GET", payload_json={}, authorize=True):
+    def _request(self, rpath, rtype="GET", payload=None, authorize=True):
```
[the remark this answers is not in the document]
Comment (gene1wood) [reply authzero.py:388]: @bkochendorfer yes, here's a good writeup on the problem with mutable default arguments : https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments

**Comment by gene1wood** [comment]
@bkochendorfer @floatingatoll I made one more change. In writing the CHANGELOG I realized that by returning `False` instead of raising an exception when a non-200 HTTP response is encountered this wouldn't be backwards compatible (there might be other stuff out there that expects an exception) so I changed the behavior back to raising an exception.

**Pull request merged**